src/cclib/method/population.py

```diff
@@ -20,6 +20,22 @@ class Population(Method):
     def __init__(self, data, progress=None, \
                  loglevel=logging.INFO, logname="Log"):
 
+        # Do we have the needed attributes in the data object?
+        # Check for getting the method  being used e.g. Mulliken Population Analysis
+        if logname == 'MPA' or logname == 'LPA':
+
+            # If the parsed data does NOT have either of the Fragment orbital overlap 
+            # or the Atomic orbital overlap matrix
+            if not (hasattr(data, "aooverlaps") \
+                 or hasattr(data, "fooverlaps") ):
+                self.required_attrs = ('mocoeffs','aooverlaps','fooverlaps','homos','nbasis')

```

Comment:
This is the tricky part. A data instance can have either `aooverlaps` or `fooverlaps`, but not both. Perhaps these should be left out of `required_attrs` and a separate check is made in the constructor.

Reply:
That makes sense. Will  do the required changes.